clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/schema/HDFSRmdSchemaSource.java

```
@@ -0,0 +1,127 @@
+ * It will make API calls on Venice controller to retrieve all RMD schemes.
+ * <p>
+ * Borrowed some codes from HDFSSchemaSource. The difference is HDFSRmdSchemaSource is created for a single store.
+ */
+@NotThreadsafe
+public class HDFSRmdSchemaSource implements RmdSchemaSource, AutoCloseable {
+  private static final Logger LOGGER = LogManager.getLogger(HDFSRmdSchemaSource.class);
+  private static final String UNDERSCORE = "_";
+  private static final String SEPARATOR = "/";
+
+  private final String storeName;
+  private final FileSystem fs;
+  private final Path schemaDir;
+
+  public HDFSRmdSchemaSource(final String schemaDir, final String storeName) throws IOException {
+    Configuration conf = new Configuration();
+    this.fs = FileSystem.get(conf);
+    this.schemaDir = new Path(schemaDir);
+    if (!fs.exists(this.schemaDir)) {
+      fs.mkdirs(this.schemaDir);
+    }
+    this.storeName = storeName;
+  }
+
+  public HDFSRmdSchemaSource(final String schemaDirSuffix) throws IOException {
+    this(schemaDirSuffix, null);
+  }
+
+  public String getPath() {
+    return schemaDir.toString();
+  }
+
+  /**
+   * This method loads related configs, gets store RMD schemas from Venice controllers, and then write them on HDFS.
+   * @throws IOException
+   * @throws IllegalStateException
+   */
+  public void loadRmdSchemasOnDisk(ControllerClient controllerClient) throws IOException, IllegalStateException {
+    LOGGER.info("Starting caching RMD schemas for {} at {}", storeName, schemaDir.toString());
+    MultiSchemaResponse.Schema[] schemas = controllerClient.getAllReplicationMetadataSchemas(storeName).getSchemas();
+    for (MultiSchemaResponse.Schema schema: schemas) {
+      // path for rmd schema is /<schemaDir>/<id>_<valueSchemaId>
+      Path schemaPath = new Path(schemaDir + SEPARATOR + schema.getId() + UNDERSCORE + schema.getRmdValueSchemaId());
+      if (!fs.exists(schemaPath)) {
```

Shall we throw an exception if the path exists? In theory, the path should be unique since we put the current timestamp part of the file path.